### ISSUE TITLE
Cannot install using install-module

### ISSUE THREAD

**Issue by kozross** [kakkun61/ghcups#7]
When I try to install this, I get the following:

```
PS C:\WINDOWS\system32> install-module -Name ghcups

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you
want to install the modules from 'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): Y
PackageManagement\Install-Package : The module 'ghcups' cannot be installed or updated because it is not a properly-formed module.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:1809 char:21
+ ...          $null = PackageManagement\Install-Package @PSBoundParameters
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Microsoft.Power....InstallPackage:InstallPackage) [Install-Package], Exception
    + FullyQualifiedErrorId : InvalidManifestModule,Test-ValidManifestModule,Microsoft.PowerShell.PackageManagement.Cmdlets.InstallPackage
```

**Comment by kakkun61** [comment]
Thanks for your trying to use ghcups and sorry for my late reply.

I cannot reproduce it.
This may be caused by the same reason with this issue https://github.com/psake/psake/issues/163.
Can you try it?

**Comment by kozross** [comment]
I'm not familiar enough with Windows to determine whether the issue is that or not. I'm running PowerShell as an administrator here, rather than as myself - does that help?

**Comment by kakkun61** [comment]
Please check your PowerShell version to type `$PSVersionTable` in the PowerShell prompt.

This is my case.

```
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.18362.145
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.145
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by spamviech** [comment]
I have the same problem (even though the error message is in german).
Output from `$PSVersionTable` is in my case:
```
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.18362.752
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.752
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

The link you posted above was confusing me, since my Powershell-knowledge is rather limited.

**Comment by kakkun61** [comment]
Hmm. I cannot reproduce it, though I tried again.
I cannot fix this issue soon. Sorry.

**Comment by spamviech** [comment]
Just tried again, error still occurs.

Full error message:
```
PS C:\WINDOWS\system32> Install-Module ghcups

Nicht vertrauenswürdiges Repository
Sie installieren die Module aus einem nicht vertrauenswürdigen Repository. Wenn Sie diesem Repository vertrauen, ändern
 Sie dessen InstallationPolicy-Wert, indem Sie das Set-PSRepository-Cmdlet ausführen. Möchten Sie die Module von
'PSGallery' wirklich installieren?
[J] Ja  [A] Ja, alle  [N] Nein  [K] Nein, keine  [H] Anhalten  [?] Hilfe (Standard ist "N"): j
PackageManagement\Install-Package : Das Modul "ghcups" kann nicht installiert oder aktualisiert werden, weil es kein
wohlgeformtes Modul ist.
In C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:1809 Zeichen:21
+ ...          $null = PackageManagement\Install-Package @PSBoundParameters
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Microsoft.Power....InstallPackage:InstallPackage) [Install-Package],
   Exception
    + FullyQualifiedErrorId : InvalidManifestModule,Test-ValidManifestModule,Microsoft.PowerShell.PackageManagement.Cm
   dlets.InstallPackage
```
Does this help?

Installed ghc through choco in the meantime, but would prefer to use this project.

**Comment by spamviech** [comment]
It is definitely an encoding error.
After enabling the beta-UTF8-support in Windows (Windows Control Panel->Region->Administrative->Change system locale -> add checkmark for Beta ut8-support -> Restart; see [here](https://stackoverflow.com/questions/388490/how-to-use-unicode-characters-in-windows-command-line/15858812#15858812) for a path to the location) the installation succeded.

Sadly, after disabling it again (since it jumbles up umlauts) I couldn't use it again (Import-Module failed).
Install-Module simply terminates, so it appears to stay installed, even though unusable.

My locale is "Deusch (Deutschland)", maybe this helps to reproduce it.

PS:
Show-Ghc for some reason only found 7.* versions, even though I have 8.10.1 (using choco) installed already.

**Comment by kakkun61** [comment]
Thank you for your investigation.
Can you tell me your terminal's code page? Run `chcp` command to know it.

> Show-Ghc for some reason only found 7.* versions, even though I have 8.10.1 (using choco) installed already.

Can you create a new issue for this?

**Comment by spamviech** [comment]
```
PS C:\WINDOWS\system32> chcp
Aktive Codepage: 850.
```

> Can you create a new issue for this?

Sure, once I find a time to restart my PC.
Forgot the exact output and it's not working with my current codepage.
Should be later today.

**Comment by kakkun61** [comment]
Neither a code page 850, [the Docker Windows container for PowerShell](https://hub.docker.com/_/microsoft-powershell) nor Deutsche locale reproduced this problem on my machine.

**Comment by spamviech** [comment]
Weird (and a unfortunate), it happens to me on two machines...

Can you point me to a good PS-Module-Tutorial page, so I can tinker with it myself once I ~~need another reason not to do my university assignments~~ find some time.

**Comment by kakkun61** [comment]
Thank you for your cooperation.

This project is my first PowerShell module and I do not actually know PowerShell modules so much.

The document I read when I started to write this module is written in Japanese.
https://tech.guitarrapc.com/entry/2013/12/03/014013

Of course I read the Microsoft documents.
https://docs.microsoft.com/en-us/powershell/scripting/developer/module/writing-a-windows-powershell-module

**Comment by spamviech** [comment]
Windows and UTF8 is a special kind of thing...

I cloned the repository into my local module path. Import-Module then gave me this error message (basically unknown characters).
```
PS C:\Users\Wolfgang> import-module ghcups
import-module : Das Modulmanifest "C:\Users\Wolfgang\Documents\WindowsPowerShell\Modules\ghcups\ghcups.psd1" konnte
nicht verarbeitet werden, da es keine gültige, eingeschränkte Sprachdatei für Windows PowerShell ist. Entfernen Sie
die für die eingeschränkte Sprache unzulässigen Elemente:
In C:\Users\Wolfgang\Documents\WindowsPowerShell\Modules\ghcups\ghcups.psd1:24 Zeichen:35
+ Author = 'Kazuki Okamoto (å²¡æœ¬å’Œæ¨¹)'
+                                   ~~~~
Unerwartetes Token "Œæ¨¹" in Ausdruck oder Anweisung.
In C:\Users\Wolfgang\Documents\WindowsPowerShell\Modules\ghcups\ghcups.psd1:24 Zeichen:35
+ Author = 'Kazuki Okamoto (å²¡æœ¬å’Œæ¨¹)'
+                                   ~
Das Hashliteral war unvollständig.
In C:\Users\Wolfgang\Documents\WindowsPowerShell\Modules\ghcups\ghcups.psd1:24 Zeichen:39
+ Author = 'Kazuki Okamoto (å²¡æœ¬å’Œæ¨¹)'
+                                       ~
Unerwartetes Token ")" in Ausdruck oder Anweisung.
In C:\Users\Wolfgang\Documents\WindowsPowerShell\Modules\ghcups\ghcups.psd1:26 Zeichen:14
+ # ã“ã®ãƒ¢ã‚¸ãƒ¥ãƒ¼ãƒ«ã®ä¼šç¤¾ã¾ãŸã¯ãƒ™ãƒ³ãƒ€ãƒ¼
+              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Unerwartetes Token "¸ãƒ¥ãƒ¼ãƒ«ã®ä¼šç¤¾ã¾ãŸã¯ãƒ™ãƒ³ãƒ€ãƒ¼" in Ausdruck oder Anweisung.
In C:\Users\Wolfgang\Documents\WindowsPowerShell\Modules\ghcups\ghcups.psd1:122 Zeichen:1
+ }
+ ~
Unerwartetes Token "}" in Ausdruck oder Anweisung.
In Zeile:1 Zeichen:1
+ import-module ghcups
+ ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (C:\Users\Wolfga...ups\ghcups.psd1:String) [Import-Module], Missing
   MemberException
    + FullyQualifiedErrorId : Modules_InvalidManifest,Microsoft.PowerShell.Commands.ImportModuleCommand
```

After changing the encoding of `ghcups.psd1` from simple `UTF-8` to `UTF-8-BOM` import-module worked like a charm.
Happy that it works now, but also annoyed that utf8 is such a pain.

**Comment by kakkun61** [comment]
Thanks a lot!

@spamviech Can you make a PR about that change?

**Comment by spamviech** [comment]
> 
> 
> Thanks a lot!
> 
> @spamviech Can you make a PR about that change?

Done: #12

**Comment by kakkun61** [comment]
The [version 3.3](https://www.powershellgallery.com/packages/ghcups/3.3) is released, which includes #12.

Please check it out.

**Comment by spamviech** [comment]
> 
> 
> The [version 3.3](https://www.powershellgallery.com/packages/ghcups/3.3) is released, which includes #12.
> 
> Please check it out.

Yes, it works now. :)

**Comment by kakkun61** [comment]
㊗️🎉
